intro-3d-2023-1/Assets/Scripts/Enemy/FiniteStateMachine/FiniteStateMachine.cs

```diff
@@ -5,15 +5,19 @@ public class FiniteStateMachine : MonoBehaviour
 {
     [Space(10)]
     [SerializeField] private Animator _anim;
-    
+
     [Space(10)]
     [SerializeField] private Transform _target;
+    
+    [Space(10)]
+    [SerializeField] private GameObject _selfObject;
 
     public Transform Target => _target;
     public NavMeshController NavMeshController => _navMeshController;
     public EnemyConfig Config => _config;
+    public GameObject SelfObject => _selfObject;
```

Comment:
Esto no es necesario.
Puedes hacer `fms.transform` o `fms.gameObject` y funciona perfecto.
También, es confuso con el propio this de C# y Unity. Para lo que lo estas utilizando, posiblemente no es necesario.

Ahora! Si esa referencia es sobre un objeto diferente (un hijo por ejemplo), entonces usa un mejor nombre que no sea `_selfObject` 😛